File: packages/sdkwork-utils-python/sdkwork_utils/bytes.py
_UNITS = ("B", "KB", "MB", "GB", "TB", "PB")


def format_bytes(value: int | float, decimals: int = 1) -> str:
    bytes_count = max(0, int(value))
    if bytes_count < 1024:
        return f"{bytes_count} B"

    size = float(bytes_count)
    unit_index = 0
    while size >= 1024 and unit_index < len(_UNITS) - 1:
        size /= 1024
        unit_index += 1

    return f"{size:.{decimals}f} {_UNITS[unit_index]}"

File: packages/sdkwork-utils-typescript/src/bytes.ts
const UNITS = ["B", "KB", "MB", "GB", "TB", "PB"] as const;

export function formatBytes(value: number, decimals = 1): string {
  const bytes = Math.max(0, Math.trunc(value));
  if (bytes < 1024) {
    return `${bytes} B`;
  }

  let size = bytes;
  let unitIndex = 0;
  while (size >= 1024 && unitIndex < UNITS.length - 1) {
    size /= 1024;
    unitIndex += 1;
  }

  return `${size.toFixed(decimals)} ${UNITS[unitIndex]}`;
}
